paper-server/src/main/java/io/papermc/paper/adventure/AdventureCodecs.java

```diff
@@ -79,6 +79,11 @@ public final class AdventureCodecs {
     public static final Codec<Component> COMPONENT_CODEC = recursive("adventure Component",  AdventureCodecs::createCodec);
     public static final StreamCodec<RegistryFriendlyByteBuf, Component> STREAM_COMPONENT_CODEC = ByteBufCodecs.fromCodecWithRegistriesTrusted(COMPONENT_CODEC);
 
+    static final Codec<ShadowColor> SHADOW_COLOR_CODEC = Codec.INT.comapFlatMap(s -> {
+        @Nullable ShadowColor value = ShadowColor.shadowColor(s);
```

Comment:
Your IDE should already be screaming at you that shadow color is never null...
There is no point in a comap flat map for something that is just an xmap.
You are also using the wrong codec, see what vanilla supports for defining shadow numbers in their own codecs (`ExtraCodecs.ARGB_COLOR_CODEC.optionalFieldOf("shadow_color").forGetter(serializer -> Optional.ofNullable(serializer.shadowColor)),`)

Reply:
Will fix that.

Reply:
Thank you for the fast fix :pray: the update is super nice for us and you beat me to it by like, half an hour (sorry if my review sounded a bit salty lol)

Reply:
Yea, no worries. I have tried to fix it. It should be okey? Its kinda hard to make it abstract for ARGBLike cause the only implementation is ShadowColorImpl. I'm okay with any criticism 👍

Reply:
Looks better tho it is still the wrong codec. Just replace Codec.INT with `ExtraCodecs.ARGB_COLOR_CODEC`.
It does not have to be an ARGBLike, that is only for the adventure setter. The getter returns a shadow color.

The mojang codec allows people to specify the color via an int[4] so we need to mirror that (hence the other base codec)

Reply:
Okay, will try that.

Reply:
Now it should be fine. 👍